readme.txt
=== Ziggeo Video for Gravity Forms ===
Contributors: oliverfriedmann, baned, carloscsz409, natashacalleia
Tags: ziggeo, video, video field, gravity forms, video submission
Requires at least: 3.0.1
Tested up to: 6.1.1
Stable tag: 1.12
Requires PHP: 5.2.4
License: GPLv3 or later
License URI: http://www.gnu.org/licenses/gpl-3.0.html

This plugin is providing an easy and simple way for you to add Ziggeo to your Gravity Forms forms through special Ziggeo fields.

= Who is this for? =

Do you need to grab video and other personal details?
Need or want to allow screen or video recording of issue that is submitted?
Form fields missing video?

If any of the above was yes, you want this plugin!

= Benefits =

Bringing the power of Ziggeo to your forms you can be sure that your forms will allow you to capture all info you need. Video can and should be easy and with this plugin that becomes true.

== Screenshots ==

1. Dashboard Setting
2. Ziggeo Fields
3. Ziggeo Video Recorder Field
4. Ziggeo Video Player Field
5. Ziggeo VideoWall Field
6. Ziggeo Templates Field
7. How data is saved in entries (you pick format)

== Installation ==
 
1. Upload plugin zip file to your plugins directory. Usually that would be to the `/wp-content/plugins/` directory
1. Activate the plugin through the 'Plugins' menu in WordPress
1. That is it or
1. Use the Plugins Add new section to find the plugin and install
 
== Frequently Asked Questions ==

= How does it work? =

The plugin combines and extends the reach of Ziggeo plugin to your Gravity Forms forms. It brings the video to your forms in an easy to use and simple manner. It allow you to add any template you have created from a single field and have video on your form.

Depending on the template you can capture screen or camera or both at the same time. All you want and so easy to get.

= Where does integration happen? =

This plugin will add the code that works in the back. It uses Gravity Forms Addon framework to bring you the addon. This addon then adds the field that you will see in the Gravity Form.

= How to use Dynamic Custom Data =

Ziggeo internally supports the ability of adding custom data to your videos. This can be anything as long as it is provided as valid JSON field. Now with form builders you might want to add custom data based on the data in the fields as well. To do this, we bring you dynamic custom data field.

* Please note that this field should not be used in combination with the custom data. You should use either `Custom Data` or `Dynamic Custom Data`.

The way you would set it up is by using key:field_id. For example if you want your JSON to be formed as:

[javascript]
{
	"first_name": "Mike",
	"last_name": "Wazowski"
}
[/javascript]

and let's say that your first name has `<input id="input_1_2" ...>` and last name has `<input id="input_1_3" ...>`. It means that we need `input_1_2` and `input_1_3` to get those values. So our field can be set as:

`first_name:input_1_2,last_name:input_1_3`

As you save your recorder field it will remember this and try to find the values. If the fields with ID are not found, the value will be saved as "" (empty string)

= How can I get some support =

We provide active support to all that have any questions or need any assistance with our plugin or our service.
To submit your questions simply go to our [Help Center](https://support.ziggeo.com/hc/en-us). Alternatively just send us an email to [support@ziggeo.com](mailto:support@ziggeo.com).

= I have an idea or suggestion =

Please go to our [WordPress forum](https://support.ziggeo.com/hc/en-us/community/topics/200753347-WordPress-plugin) and add your suggestion within it. This allows everyone to see and vote on it and us to determine what should be next.

== Upgrade Notice ==

= 1.12 =
* Improvement: Changed the way lazy loading is done. Mostly so that the code is in one location, instead of having duplicate codes in few places.

== Changelog ==

Please check CHANGELOG.md found in the plugin files. This file contains all of the log entries for past versions.

CHANGELOG.md
This file contains the change log info for the `Ziggeo - Gravity Forms` Bridge plugin


=======

= 1.11 =
* Added: Added support for WP Core plugin Lazyload feature
* Improvement: Removed dependency on jQuery which could lead to console error in the preview page

= 1.10 =
* Added: Hooks for verified event
* Added: Dynamic custom data support, allowing you to create custom data based on the fields on your Gravity Forms form.

= 1.9 =
* Added the option to set tags to video based on other fields on form. The same happens once the video is verified.
* Added the check for the conditional logic so that notices are not outputted in cases when your server does not hide them (you should if not dev site ;) ).
* Improvement: API calls are now using V2 flavor.
* Improvement: Submenu calls are now checking if core plugin is relatively fresh and available to support the call. otherwise it does not use it.
* Improvement: Added a check when the plugin is loading to see if core plugin is available or not. Instead of skipping loading silently, we now show a warning about this in admin dashboard.

= 1.8 =
* Utilizing the function Ziggeo Core plugin provides to add the page as addon
* Removed the notification that would be made each time you request the plugin and the Videowalls plugin is not installed or activated. Now fires only in editor. Also shows notice in the form builder that it is missing.

= 1.7.2 =
* Fixed videowall assets loading

= 1.7.1 =
* Fixed the issue with the templates submission where if you used recorder or player through a template you might not get the value in because of new validation that was added.

= 1.6 =
* Added the option to select what type of value you want to get once the video is recorded, which is made available through dashboard page under Ziggeo Video > Ziggeo Video for Gravity Forms.
* Also added a filter to allow this to change this through your own code regardless of the setting
* Any time video is recorded it will add +1 new video in the video list

= 1.5 =
* Added a check to avoid error if the application token was not provided. It also helps if core plugin was not installed.

= 1.4 =
* Fixed an issue caused by the space character in video recorder.

= 1.3 =
* Added support for No Conflict mode of Gravity Forms editor.

= 1.2. =
* Improved support for core plugin
* Field for templates have been modified to provide better support for all templates
* Additional fields have been added (Recorder, Player and VideoWalls)
* All buttons are now within Ziggeo fields group of fields

= 1.1 =
* Upgrading versions
* Support for new Ziggeo core plugin (v2.0)

= 1.0 =
* Initial commit

core/run.php
<?php

//
//	This file represents the integration module for Gravity Forms and Ziggeo
//

// Index
//	1. Hooks
//		1.1. ziggeo_list_integration
//		1.2. gform_loaded
//	2. Functionality
//		2.1. ziggeogravityforms_get_version()
//		2.2. ziggeogravityforms_init()
//		2.3. ziggeogravityforms_run()


//Checking if WP is running or if this is a direct call..
defined('ABSPATH') or die();

/////////////////////////////////////////////////
// 1. HOOKS                                    //
/////////////////////////////////////////////////

	//Show the entry in the integrations panel
	add_filter('ziggeo_list_integration', function($integrations) {

		$current = array(
			//This section is related to the plugin that we are combining with the Ziggeo, not the plugin/module that does it
			'integration_title'		=> 'Gravity Forms', //Name of the plugin
			'integration_origin'	=> 'http://www.gravityforms.com/', //Where you can download it from

			//This section is related to the plugin or module that is making the connection between Ziggeo and the other plugin.
			'title'					=> 'Ziggeo Video for Gravity Forms', //the name of the module
			'author'				=> 'Ziggeo', //the name of the author
			'author_url'			=> 'https://ziggeo.com/', //URL for author website
			'message'				=> 'Add video to your forms', //Any sort of message to show to customers
			'status'				=> true, //Is it turned on or off?
			'slug'					=> 'ziggeo-video-for-gravity-forms', //slug of the module
			//URL to image (not path). Can be of the original plugin, or the bridge
			'logo'					=> ZIGGEOGRAVITYFORMS_ROOT_URL . 'assets/images/logo.png',
			'version'				=> ZIGGEOGRAVITYFORMS_VERSION
		);

		//Check current Ziggeo version
		if(ziggeogravityforms_run() === true) {
			$current['status'] = true;
		}
		else {
			$current['status'] = false;
		}

		$integrations[] = $current;

		return $integrations;
	});

	//plugins_loaded is one hook too late for GravityForms to have the field present
	add_action('gform_loaded', function() {
		ziggeogravityforms_run();
	}, 5);




/////////////////////////////////////////////////
// 2. FUNCTIONALITY                            //
/////////////////////////////////////////////////

	//Checks if the Gravity Forms exists and returns the version of it
	function ziggeogravityforms_get_version() {
		if(class_exists('GFForms')) {
			return GFForms::$version;
		}

		return 0;
	}

	//We add all of the hooks we need
	function ziggeogravityforms_init() {

		$options = ziggeogravityforms_get_plugin_options();

		//Add our addon in such a way that it only runs if Gravity Forms was initiallized properly
		//We need the addon framework to be available
		if( !method_exists('GFForms', 'include_addon_framework') ) {
			return;
		}

		//Include the required codes
		require_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'core/simplifiers.php');

		//include the addon classes
		include_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'extend/ziggeo-addon-video-recorder.php');
		GFAddOn::register( 'Ziggeo_GF_VideoRecorder_Addon' );

		include_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'extend/ziggeo-addon-video-player.php');
		GFAddOn::register( 'Ziggeo_GF_VideoPlayer_Addon' );

		include_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'extend/ziggeo-addon-ziggeo-templates.php');
		GFAddOn::register( 'Ziggeo_GF_ZiggeoTemplates_Addon' );

		//This will not work if the videowalls-for-ziggeo is not installed.
		//Get from: https://wordpress.org/plugins/videowalls-for-ziggeo/
		include_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'extend/ziggeo-addon-video-wall.php');
		GFAddOn::register( 'Ziggeo_GF_VideoWall_Addon' );


		include_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'core/assets.php');
		include_once(ZIGGEOGRAVITYFORMS_ROOT_PATH . 'core/hooks.php');

		/*add_action('gform_entry_detail_content_before', function($form, $lead) {

			//We check the form fields for our own field types and check the content submitted on them
			//for($i = 0, $c = count($form['fields']); $i < $c; $i++) {
			//	if($form['fields'][$i]['type'] === 'ZiggeoRecorder') {
			//		$lead[$form['fields'][$i]['id']] = ziggeo_p_content_filter($lead[$form['fields'][$i]['id']], $lead[$form['fields'][$i]['id']]);
			//	}
			//}
			//The code works to find things, however not to change them
			//We could use this to pass JS function that would replace the value in preview..

			//
		}, 10, 2);*/

	}

	//Function that we use to run the module 
	function ziggeogravityforms_run() {

		//Needed during activation of the plugin
		if(!function_exists('ziggeo_get_version')) {
			add_action( 'admin_notices', function() {
				?>
				<div class="error notice">
					<p><?php _e( 'Please install <a href="https://wordpress.org/plugins/ziggeo/">Ziggeo plugin</a>. It is required for this plugin (Ziggeo Video for Gravity Forms) to work properly!', 'ziggeogravityforms' ); ?></p>
				</div>
				<?php
			});

			return false;
		}

		//Check current Ziggeo version
		if( version_compare(ziggeo_get_version(), '2.0') >= 0 ) {
			//check the Gravity Forms version
			if(version_compare(ziggeogravityforms_get_version(), '1.9') >= 0) {
				if(ziggeo_integration_is_enabled('ziggeo-video-for-gravity-forms')) {
					ziggeogravityforms_init();
					return true;
				}
			}
		}

		return false;
	}


?>